### ISSUE TITLE
Request:  Sync sshPublicKey from LDAP into Gitlab

### ISSUE THREAD

**Issue by sunrez** [Adambean/gitlab-ce-ldap-sync#11]
Hi,

First, thank you so much for taking the time to create this very useful solution to sync Gitlab against an LDAP server!

I had a feature request:  Would you consider adding support to pull available sshPublicKeys from a user LDAP entry and sync that into Gitlab? 

I AM able to get OpenSSH to pull sshPublicKeys from OpenLDAP using AuthorizedKeyCommands (https://serverfault.com/questions/653792/ssh-key-authentication-using-ldap , https://askubuntu.com/questions/776700/ssh-ldap-authorizedkeyscommand ) 

but unfortunately GitLab doesn't honor this.

That feature is also a Gitlab-EE feature ("sync_ssh_keys").

**Comment by Adambean** [comment]
This is a good idea.

The configuration will need an option as how to handle extra keys, e.g. those that a user has added to Gitlab themselves but not found in their directory object. (Probably a rare scenario.)

**Comment by Adambean** [comment]
I don't think this is required any more? I've noticed for some time that GitLab appears to pull `sshPublicKey` values from LDAP automatically.

<img width="1907" height="454" alt="Image" src="https://github.com/user-attachments/assets/ca0eb4c4-f412-4194-aec8-87f8a2a0d171" />
